src/test/java/com/juniorstart/juniorstart/service/ProjectServiceTest.java

```diff
@@ -0,0 +1,228 @@
+
+        PagedResponse<ProjectResponse> response = projectService.getAllProjects(0 , 10);
+        assertTrue(response.getContent().get(0).getBody().contains(project.getBody()));
+        assertTrue(response.getContent().get(0).getDescription().contains(project.getDescription()));
+        assertTrue(response.getContent().get(0).getName().contains(project.getName()));
+        assertTrue(response.getContent().get(0).getTitle().contains(project.getTitle()));
+
+        verify(projectRepository, times(1)).findAll(isA(Pageable.class));
+        verify(userDao, times(1)).findByPublicId(anyLong());
+    }
+
+
+    @Test
+    public void should_create_new_project() throws Exception {
+
+        when(userDao.findById(ArgumentMatchers.any(UUID.class))).thenReturn(Optional.of(user));
+        when(projectRepository.findByName(ArgumentMatchers.any(String.class))).thenReturn(Optional.empty());
+        when(projectRepository.save(ArgumentMatchers.any(Project.class))).thenReturn(project);
+        Project response = projectService.createProject(userPrincipal, projectRequest);
+        assertTrue(response.getBody().contains(project.getBody()));
+        assertTrue(response.getName().contains(project.getName()));
+        assertEquals(response.getRecruiting(), project.getRecruiting());
+        verify(userDao, times(1)).findById(any(UUID.class));
+        verify(projectRepository, times(1)).findByName(anyString());
+        verify(projectRepository, times(1)).save(any(Project.class));
+    }
+
+    @Test
+    public void should_update_project() throws Exception {
+       when(userDao.findById(any(UUID.class))).thenReturn(Optional.of(user));
+       when(projectRepository.findById(anyLong())).thenReturn(Optional.of(project));
+       when(projectRepository.save(any(Project.class))).thenReturn(project);
+
+       ProjectResponse response = projectService.updateProject(userPrincipal, project.getId(), projectRequest);
+
+        assertTrue(response.getName().contains(project.getName()));
+        assertTrue(response.getTitle().contains(project.getTitle()));
+        assertTrue(response.getDescription().contains(project.getDescription()));
+
+        verify(userDao, times(1)).findById(any(UUID.class));
+        verify(projectRepository, times(1)).findById(anyLong());
+        verify(projectRepository, times(1)).save(any(Project.class));
+    }
+
+    @Test
+    public void should_delete_project() throws Exception {
+        when(projectRepository.findById(anyLong())).thenReturn(Optional.of(project));
+        when(userDao.findByPrivateId(any(UUID.class))).thenReturn(Optional.of(user));
+        ResponseEntity<?> response = projectService.deleteProject(userPrincipal, project.getId());
+        assertEquals(response.getStatusCodeValue(), HTTPResponse.SC_OK);
+        verify(projectRepository, times(1)).findById(anyLong());
+        verify(userDao, times(1)).findByPrivateId(any(UUID.class));
+    }
```

Comment:
should_update_project() napisany czytelnie ładnie should_delete_project() naźgane. Zobacz jak widać różnice. O to chodzi że jak na to spojrzysz za miesac albo 2 to żebyś od razu mógł sie doczytać.